lib/Browser.js

```diff
@@ -81,3 +147,61 @@ class TaskQueue {
     return result;
   }
 }
+
+class Target {
+  /**
+   * @param {Browser} browser
+   * @param {string} targetId
+   * @param {string} type
+   */
+  constructor(browser, targetId, type) {
+    this._browser = browser;
+    this._targetId = targetId;
+    this._type = type;
+    /** @type {Promise<Page>} */
+    this._pagePromise = null;
+  }
+
+  /**
+   * @return {Promise<{targetId: string, type: string, title: string, url: string, attached: boolean, type: string}>}
+   */
+  async _info() {
```

Comment:
you should create target with targetInfo and make this sync.
You'll also need to listen to `targetInfoChanged` event.